sdk/tracing/azure-core-tracing-opencensus/src/test/java/com/azure/core/tracing/opencensus/OpenCensusTracerTest.java

```diff
@@ -36,13 +77,111 @@ public void startSpanNullPointerException() {
     }
 
     @Test
-    public void startSpanReturnsNewSpanTest() {
+    public void startSpanParentContextFlowTest() {
+        // Arrange
+        SpanId parentSpanId = parentSpan.getContext().getSpanId();
+
+        // Act
+        final Context updatedContext = openCensusTracer.start(METHOD_NAME, tracingContext);
+
+        // Assert
+        assertSpanWithExplicitParent(updatedContext, parentSpanId);
+        RecordEventsSpanImpl recordEventsSpan = (RecordEventsSpanImpl) updatedContext.getData(OPENCENSUS_SPAN_KEY).get();
+        Assert.assertNull(recordEventsSpan.getKind());
+    }
+
+    @Test
+    public void startSpanTestNoUserParent() {
         // Act
         final Context updatedContext = openCensusTracer.start(METHOD_NAME, Context.NONE);
 
         // Assert
         Assert.assertNotNull(updatedContext.getData(OPENCENSUS_SPAN_KEY));
-        assertTrue(updatedContext.getData(OPENCENSUS_SPAN_KEY).get() instanceof Span);
+
+        //verify still get a valid span implementation
+        Assert.assertTrue(updatedContext.getData(OPENCENSUS_SPAN_KEY).get() instanceof RecordEventsSpanImpl);
+        RecordEventsSpanImpl recordEventsSpan = (RecordEventsSpanImpl) updatedContext.getData(OPENCENSUS_SPAN_KEY).get();
+
+        Assert.assertEquals(METHOD_NAME, recordEventsSpan.getName());
+        Assert.assertFalse(recordEventsSpan.toSpanData().getHasRemoteParent());
+        Assert.assertNotNull(recordEventsSpan.toSpanData().getParentSpanId());
+    }
+
+    @Test
+    public void startSpanProcessKindSend() {
+        // Arrange
+        tracer.spanBuilder(OPENCENSUS_SPAN_KEY).startScopedSpan();
+        SpanId parentSpanId = parentSpan.getContext().getSpanId();
+        // Add additional metadata to spans for SEND
+        Context traceContext = tracingContext.addData(ENTITY_PATH, ENTITY_PATH_VALUE).addData(HOST_NAME, HOSTNAME_VALUE);
+
+        // Act
+        final Context updatedContext = openCensusTracer.start(METHOD_NAME, traceContext, ProcessKind.SEND);
+
+        // Assert
+        // verify span created with explicit parent when for Process Kind SEND
+        assertSpanWithExplicitParent(updatedContext, parentSpanId);
+        RecordEventsSpanImpl recordEventsSpan = (RecordEventsSpanImpl) updatedContext.getData(OPENCENSUS_SPAN_KEY).get();
+        Assert.assertEquals(Span.Kind.CLIENT, recordEventsSpan.getKind());
+
+        // verify span attributes
+        Map<String, AttributeValue> attributeMap = recordEventsSpan.toSpanData().getAttributes().getAttributeMap();
+        Assert.assertEquals(attributeMap.get(COMPONENT), AttributeValue.stringAttributeValue(COMPONENT_VALUE));
+        Assert.assertEquals(attributeMap.get(MESSAGE_BUS_DESTINATION), AttributeValue.stringAttributeValue(ENTITY_PATH_VALUE));
+        Assert.assertEquals(attributeMap.get(PEER_ENDPOINT), AttributeValue.stringAttributeValue(HOSTNAME_VALUE));
+    }
+
+    @Test
+    public void startSpanProcessKindReceive() {
```

Comment:
I recall we decided against instrumenting Receive call

Reply:
Receive is the message span. 
I think with eventhubs we wanted to call it this way as "message" didn't macth any of th existing terms we used in the package.